ls1intum/Artemis
# Advanced exam checklist instructor dashboard

### Checklist

### Motivation and Context


Exams in Artemis provide instructors with a lot of options and configuration possibilities. To make it easier for instructors we implemented the exam checklist, which displays all the necessary steps when conducting an exam. Now in this followup we want to enhance it with dynamic checks. 

### Description
I enhanced the existing exam instructor checklist with dynamic checks.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration -> Exams -> Click on an Exam title
3. Make sure that all the new checks correspond with the current status of the exam

### Test Coverage



exam-detail.component.ts : 100%
exam-checklist-check.component.ts : 100% 

### Screenshots


1. 
![Bildschirmfoto 2021-02-07 um 18 28 21](https://user-images.githubusercontent.com/42205689/107155961-d3c92200-697b-11eb-9aea-8cca9a5f8f21.png)
2. 
<img width="984" alt="Bildschirmfoto 2021-02-08 um 09 21 21" src="https://user-images.githubusercontent.com/42205689/107193380-15df7b80-69ef-11eb-8ef1-e31630473f50.png">
3. 
<img width="992" alt="Bildschirmfoto 2021-02-08 um 09 21 35" src="https://user-images.githubusercontent.com/42205689/107193464-2b54a580-69ef-11eb-9409-d65d76e37ebd.png">
